tests/nightly/test_large_vector.py

```diff
@@ -708,6 +708,182 @@ def test_full():
     assert a[-1] == 3
 
 
+def test_astype():
+    x = create_vector(size=LARGE_X//4)
+    x = nd.tile(x, 4)
+    y = x.astype('int32')
+    assert y.dtype == np.int32
+    assert y[-1] == LARGE_X//4-1
+
+
+def test_cast():
+    x = create_vector(size=LARGE_X//4)
+    x = nd.tile(x, 4)
+    y = nd.cast(x, np.int32)
+    assert y.dtype == np.int32
+    assert y[-1] == LARGE_X//4-1
+
+
+def test_repeat():
+    x = create_vector(size=LARGE_X//2)
+    y = nd.repeat(x, repeats=2, axis = 0)
+    assert y.shape[0] == LARGE_X
+    assert y[1] == 0
+    assert y[LARGE_X-1] == LARGE_X//2-1
+
+
+def create_input_for_rounding_ops():
+    inp = nd.arange(-LARGE_X//2, LARGE_X//2, dtype=np.float64)
+    inp = inp/2
+    return inp
+
+
+def test_ceil():
+    x = create_input_for_rounding_ops()
+    y = nd.ceil(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == 0
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 1
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_fix():
+    x = create_input_for_rounding_ops()
+    y = nd.fix(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == 0
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 0
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_floor():
+    x = create_input_for_rounding_ops()
+    y = nd.floor(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == -1
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 0
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_rint():
+    x = create_input_for_rounding_ops()
+    y = nd.rint(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == -1
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 0
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_round():
+    x = create_input_for_rounding_ops()
+    y = nd.round(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == -1
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 1
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_trunc():
+    x = create_input_for_rounding_ops()
+    y = nd.trunc(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == 0
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 0
```

Comment:
looks like a bunch of repeated code that can be reused in these 5 tests.

Reply:
all differenct assetions that explain the difference between each of the rounding functions

Reply:
this is a nice to have, not blocking on this. But this pattern can be refactored: create input, execute op, check output at specific indexes ( which happen to be same among all tests).